lib/shimmer.js

```diff
@@ -613,7 +618,12 @@ function _instrumentOnResolved(agent, moduleName, resolvedFilepath) {
 
   const shim = shims.createShimFromType(instrumentation.type, agent, moduleName, resolvedFilepath)
 
-  trackInstrumentationUsage(agent, shim, moduleName, NAMES.FEATURES.INSTRUMENTATION.ON_RESOLVED)
+  trackInstrumentationUsage(
+    agent,
+    shim,
+    instrumentation.friendlyModuleName || moduleName,
```

Comment:
Forgot to add a unit test for this and Ln#520, will add

Reply:
this brings up an interesting use case that I don't think we confirmed or not but we have some onResolved hooks and I'm not sure if they will continue to work if you import said module.  I think the one that sticks out is [aws-sdk v3](https://github.com/newrelic/node-newrelic-aws-sdk/blob/main/nr-hooks.js)

Reply:
So I did a sanity check, and I was able to swap between the 9.0.3 version of the agent and the `main` branch with the esm-loader, and still had aws-sdk instrumentation come up in the APM dashboard. I used a [slightly modified version](https://gist.github.com/jmartin4563/7a19fa473704e6bab24ee49b81144aae) of the script you sent me earlier.